### PR TITLE
refactor: add LinkButton, deprecate Button

### PR DESCRIPTION
**Pull request by kurtaking**
Signed-off-by: Kurt King <kurtaking@gmail.com>

## Hey, I just made a Pull Request!

This PR is related to issue https://github.com/backstage/backstage/issues/14889 - Bug Report: Button component is confusing.

- Adds new `LinkButton` component
- Deprecates `Button` component

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by kurtaking** [comment]
@Rugvip, I have created a separate `LinkButton` component and referenced it in the deprecation message in `Button`. Let me know if there is a better way to do this. I saw one example (`DismissbleBanner`) where it looks like the deprecation and replacement were in the same file.

Separately, would we want to include or follow up with a PR replacing all references of `Button` in `core-components`, `/packages`?

File: packages/core-components/src/components/Link/Link.stories.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020 The Backstage Authors
+ * Copyright 2022 The Backstage Authors
```
Comment (Rugvip) [inline packages/core-components/src/components/Link/Link.stories.tsx:2]: Keep these as they were. It's for the creation of the file.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 

I think it's best to remove the existing `Button` files, i.e. rename them, to make sure it's identified as a move by git and we keep the history and avoid the duplicate code. Then at the bottom of the new `LinkButton` we can add deprecated aliases, e.g.

```tsx
/** ... */
export const Button = LinkButton
/** ... */
export type ButtonProps = LinkButtonProps
```

Yep once this is in I think it makes sense to update our own usage to use the `LinkButton` instead.

**Comment by kurtaking** [comment]
Today is my first day back from vacation. I will pick this back up this week.

**Comment by kurtaking** [comment]
> Thank you! 👍
> 
> I think it's best to remove the existing `Button` files, i.e. rename them, to make sure it's identified as a move by git and we keep the history and avoid the duplicate code. Then at the bottom of the new `LinkButton` we can add deprecated aliases, e.g.
> 
> ```tsx
> /** ... */
> export const Button = LinkButton
> /** ... */
> export type ButtonProps = LinkButtonProps
> ```
> 
> Yep once this is in I think it makes sense to update our own usage to use the `LinkButton` instead.

I am getting the PR up to date with master and then will look at renaming everything. Thanks for the response!

**Comment by kurtaking** [comment]
@Rugvip, looking to get a review at your earliest convenience.

File: packages/core-components/src/components/LinkButton/LinkButton.tsx [hunk 2]
```diff
@@ -43,9 +43,17 @@ const LinkWrapper = React.forwardRef<any, LinkProps>((props, ref) => (
...
+
+/**
+ * @deprecated use LinkButton instead
+ */
+export const Button = LinkButton;
+
+/**
+ * @deprecated use LinkButtonProps instead
+ */
+export type ButtonProps = LinkButtonProps;
```
Comment (Rugvip) [inline packages/core-components/src/components/LinkButton/LinkButton.tsx:59]: These two need to be marked with `@public`

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 

Just a minor nit to clean up the API report

File: packages/core-components/src/components/LinkButton/LinkButton.tsx (hunk 2 above)
Comment (kurtaking) [reply packages/core-components/src/components/LinkButton/LinkButton.tsx]: Okay, got that taken care of.

**Review by Rugvip — APPROVED** [review]
Thank you! 👍 

Let's :shipit: 🎉

**Pull request merged**